### PR TITLE
ROS Drivers for OVC5

### PR DESCRIPTION
**Pull request by gbalke**
This PR will introduce ROS1/2 drivers for OVC. My intent is to use https://github.com/romainreignier/share_ros1_ros2_lib_demo as an example for building the libs in a convenient way.

TODO:

- [x] Extract common code into a library
- [x] Write ROS2 driver
- [x] Figure out how to make both ROS and `cv::imshow` happy with output of `cv::cvtColor`
- [x] Add configuration hooks to allow host to device configuration
- [x] Dynamic Reconfigure for ROS1
- [x] Parameters for ROS2

### CODE REVIEW

**Comment by luca-della-vedova** [comment]
I was halfway through reviewing when I saw that quite a bit of the diff is just changing style for parenthesis from

```
void fun()
{
   // stuff
}
```

to

```
void fun() {
    // stuff
}
```

Regardless of which way is better (I personally prefer the first but I know it's a pretty trivial matter) I think the codebase is pretty consistent in the newline style and this style change makes the diff a lot larger, as well as introduce inconsistencies between all the files that have been touched by this PR and all of those that were not.

**Comment by gbalke** [comment]
> I was halfway through reviewing when I saw that quite a bit of the diff is just changing style for parenthesis from
> 
> ```
> void fun()
> {
>    // stuff
> }
> ```
> 
> to
> 
> ```
> void fun() {
>     // stuff
> }
> ```
> 
> Regardless of which way is better (I personally prefer the first but I know it's a pretty trivial matter) I think the codebase is pretty consistent in the newline style and this style change makes the diff a lot larger, as well as introduce inconsistencies between all the files that have been touched by this PR and all of those that were not.

I use FormatCode from https://github.com/google/vim-codefmt. I'll try to adjust the formatter to match this style requirement and give you a copy of what I get. I'd rather just always run code through it rather than having to have formatting "holy wars" :smile:

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+BasedOnStyle: Google
+---
+Language: Cpp
+BreakBeforeBraces: Allman
+AccessModifierOffset: -2
```
Comment (gbalke) [inline .clang-format:6]: @luca-della-vedova please let me know if there's anything else I should adjust.

**Comment by gbalke** [comment]
Waiting on #48 to merge.

**Comment by gbalke** [comment]
I'll give this a test tomorrow before merging.

**Pull request merged**